src/components/Shared/RecordRemarks.js

```diff
@@ -0,0 +1,162 @@
+import React, { useContext, useState } from 'react'
+import Grid from '@mui/system/Unstable_Grid/Grid'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Box, IconButton, TableBody, Table, TableCell, TableContainer, TableRow } from '@mui/material'
+import Icon from 'src/@core/components/icon'
+import moment from 'moment'
+
+// import Table from 'src/components/Shared/Table'
+
+import { useResourceQuery } from 'src/hooks/resource'
+import RecordRemarksForm from './RecordRemarksForm'
+import { useWindow } from 'src/windows'
+import DeleteDialog from './DeleteDialog'
+
+const RecordRemarks = ({ recordId, resourceId, expanded }) => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { stack } = useWindow()
+  const [deleteDialogOpen, setDeleteDialogOpen] = useState([false, {}])
+  const userId = JSON.parse(window.sessionStorage.getItem('userData'))?.userId
+
+  const fetchGridData = () => {
+    var parameters = `_resourceId=${resourceId}&_masterRef=${recordId}`
+
+    return getRequest({
+      extension: SystemRepository.RecordRemarks.qry,
+      parameters: parameters
+    })
+  }
+
+  const date = date => {
+    const eventDate = moment(date)
+    const formattedDate = eventDate.calendar()
+
+    return formattedDate
+  }
+
+  const {
+    query: { data },
+    labels: labels,
+    access
+  } = useResourceQuery({
+    enabled: !!recordId,
+    datasetId: ResourceIds.RecordRemarks,
+    queryFn: fetchGridData,
+    endpointId: SystemRepository.RecordRemarks.qry
+  })
+
+  const onEdit = obj => {
+    stack({
+      Component: RecordRemarksForm,
+      props: {
+        data: obj,
+        labels,
+        userId
+      },
+      width: 800,
+      height: 300,
+      title: labels.resourceRecordRemarks
+    })
+  }
+
+  console.log(data)
+
+  const dataTable = {
+    ...data,
+    list: (data?.list || []).map(({ notes, eventDate, userName, ...rest }) => ({
+      notes: '<b>' + userName + '</b> ' + date(eventDate) + '<br/><br />' + notes,
+      ...rest
+    }))
+  }
+  console.log(dataTable)
+
+  const onDelete = async obj => {
+    await postRequest({
+      extension: SystemRepository.RecordRemarks.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
+  }
+
+  return (
+    <Box>
+      <RecordRemarksForm
+        userId={userId}
+        resourceId={resourceId}
+        labels={labels}
+        masterRef={recordId}
+        maxAccess={access}
+        seqNo={data?.list?.length + 1}
```

Comment:
no, you get the maximum seqNo and you add 1 to it. not the length
example : I have 3 rows with seqNo 1, 2 and 3.
If I delete the row with seqNo = 2 -> I will have 2rows with seqNo 1 and 3.
The list length is 2 so when I need to add another note, the new seqNo is 3 which is already there, it will override it.